test/kvao/ttl.suite.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+var should = require('should');
+var helpers = require('./_helpers');
+var Promise = require('bluebird');
+
+module.exports = function(dataSourceFactory, connectorCapabilities) {
+  describe('ttl', function() {
+    var CacheItem;
+    beforeEach(function unpackContext() {
+      CacheItem = helpers.givenCacheItem(dataSourceFactory);
+    });
+
+    context('existing key with expire before expiration time', function() {
+      it('returns ttl - Callback API', function(done) {
+        CacheItem.set('a-key', 'a-value', 10, function(err) {
+          if (err) return done(err);
+          setTimeout(function() {
```

Comment:
I think it might not be needed if within works as expected. I needed it cause I used `lessThan(10)` and the timeout guarantee some time has passed (ie. doesn't return 10 again). I should've used `lessThanOrEqual(10)`, but your suggestion of `within(0, 10)` is even better.